src/pylint_odoo/checkers/import_checker.py

```diff
@@ -0,0 +1,128 @@
+import re
+from typing import Union
+
+from astroid import Import, ImportFrom, Module
+from astroid.modutils import modpath_from_file
+from pylint.checkers.utils import only_required_for_messages
+from pylint.lint import PyLinter
+
+from .odoo_base_checker import OdooBaseChecker
+
+ODOO_MSGS = {
+    # C->convention R->refactor W->warning E->error F->fatal
+    "E8401": (
+        "This python module is not imported and has no effect",
+        "unused-module",
+        "",
+    ),
+}
+
+EXCLUDED = re.compile("__manifest__.py$|__openerp__.py$|__init__.py$|tests/|migrations/|tests\\\\|migrations\\\\")
+
+
+class ImportChecker(OdooBaseChecker):
+    msgs = ODOO_MSGS
+    name = "odoolint"
+
+    def __init__(self, linter: PyLinter):
+        self.imports = set()
+        self.modules = set()
+
+        super().__init__(linter)
+
+    @staticmethod
+    def get_module_from_node(node, max_attempts: int = 15) -> Union[None, Module]:
+        """Get the module a node belongs to.
+
+        :param node: Node that belongs to the module to be extracted.
+        :param int max_attempts: Amount of attempts that will be made to obtain the module. Nodes that are
+        nested deeper than max_attempts won't be found.
+
+        :returns: Module if found, otherwise None.
+        """
+        attempts = 0
+        while attempts < max_attempts:
+            if not getattr(node, "parent", False):
+                return None
+
+            if isinstance(node.parent, Module):
+                return node.parent
+
+            node = node.parent
+            attempts += 1
+
+        return None
+
+    def store_imported_modules(self, node):
+        """Store all modules that are imported by 'from x import y' and 'import x,y,z' statements.
+
+        Relative paths are taken into account for ImportFrom nodes. For example, the following statements
+        would import the following modules (consider the file which contains these statements is module.models.partner):
+
+        from ..wizard import cap, hello --> module.wizard, module.wizard.cap, module.wizard.hello
```

Comment:
Indent, to indicate it's code

Reply:
I consider indenting code is cleaner. CUrrent approach uses several backticks on every line.